### ISSUE TITLE
Handling async calls in queries and mutations

### ISSUE THREAD

**Issue by dimitkos** [graphql-dotnet/graphql-dotnet#3832]
Hello, I am using .net 7  and graphql version 7.6.0 i am trying to understand how to handle async calls into queries and mutations.

For example, I have 

```
public interface IEmployeeRepository
{
   Task<List<Employee>> GetAllEmployees();
}
```

and I trying to create a query 

```
 public class EmployeeQuery : ObjectGraphType
 {

     public EmployeeQuery (IEmployeeRepository repository)
     {
         FieldAsync<ListGraphType<ReservationType>>("employees",
         resolve: async (context) => await repository.GetAllEmployees();
     }
 }
```
 
 but with this option, I get an obsolete warning.
 
 Could you provide me an example of query and mutation?
 
 Thanks in advance

**Comment by gao-artur** [comment]
It should be
```c#
 public class EmployeeQuery : ObjectGraphType
 {
     public EmployeeQuery (IEmployeeRepository repository)
     {
         Field<ListGraphType<ReservationType>>("employees")
             .ResolveAsync(async (context) => await repository.GetAllEmployees());
     }
 }
```

Try the recently released v7.7.1, which will help you rewrite the obsolete API automatically.

**Comment by Shane32** [comment]
Please ask if you have further questions
